File: 8kyu/is-it-a-palindrome/is-it-a-palindrome.go
package kata

import "strings"

func IsPalindrome(str string) bool {
	if len(str) < 2{
		return true
	}

	left, right := 0, len(str)-1

	for left < right {
		if strings.ToLower(str)[left] != strings.ToLower(str)[right] {
			return false
		}
		left++
		right--
	}
 return true
}

File: 8kyu/is-it-a-palindrome/is-it-a-palindrome.js
function isPalindrome(x) {
  x.toLowerCase()
  if (x.length < 2) return true

  let [left, right] = [0, x.length - 1]

  while (left < right) {
    if (x.toLowerCase()[left] !== x.toLowerCase()[right]) return false
    left += 1
    right -= 1
  }

  return true
}


console.log(isPalindrome(""))
console.log(isPalindrome("a"))